src/SDK/MonitoringApplianceEmulator/ViewModels/MetricChartPropertyControlViewModel.cs

```diff
@@ -0,0 +1,213 @@
+        /// <param name="analysisServicesFactory">The analysis services factory</param>
+        /// <param name="metricChartAlertProperty">The metric chart alert property that should be displayed.</param>
+        /// <param name="tracer">The tracer</param>
+        public MetricChartPropertyControlViewModel(
+            MetricChartAlertProperty metricChartAlertProperty,
+            IAnalysisServicesFactory analysisServicesFactory,
+            ITracer tracer)
+        {
+            this.Title = metricChartAlertProperty.DisplayName;
+            this.analysisServicesFactory = analysisServicesFactory;
+            this.tracer = tracer;
+
+            // Start a task to read the metric values
+            this.loadChartTask = new ObservableTask<ChartValues<DateTimePoint>>(
+                this.ReadChartValues(metricChartAlertProperty),
+                tracer,
+                this.DisplayChartValues);
+        }
+
+        /// <summary>
+        /// Gets the chart title.
+        /// </summary>
+        public string Title { get; }
+
+        /// <summary>
+        /// Gets the series collection.
+        /// </summary>
+        public SeriesCollection SeriesCollection
+        {
+            get
+            {
+                return this.seriesCollection;
+            }
+
+            private set
+            {
+                this.seriesCollection = value;
+                this.OnPropertyChanged();
+            }
+        }
+
+        /// <summary>
+        /// Gets the X axis formatter.
+        /// </summary>
+        public Func<double, string> XAxisFormatter
+        {
+            get
+            {
+                return this.xAxisFormatter;
+            }
+
+            private set
+            {
+                this.xAxisFormatter = value;
+                this.OnPropertyChanged();
+            }
+        }
+
+        /// <summary>
+        /// Gets the Y axis formatter.
+        /// </summary>
+        public Func<double, string> YAxisFormatter
+        {
+            get
+            {
+                return this.yAxisFormatter;
+            }
+
+            private set
+            {
+                this.yAxisFormatter = value;
+                this.OnPropertyChanged();
+            }
+        }
+
+        /// <summary>
+        /// Asynchronously load the metric chart
+        /// </summary>
+        /// <param name="property">The metric chart alert property</param>
+        /// <returns>A <see cref="Task"/>, running the current operation</returns>
+        private async Task<ChartValues<DateTimePoint>> ReadChartValues(MetricChartAlertProperty property)
+        {
+            CancellationToken cancellationToken = CancellationToken.None;
+
+            // Verify start/end times
+            DateTime startTime = property.StartTimeUtc ?? throw new ApplicationException("Start time cannot be null");
+            DateTime endTime = property.EndTimeUtc ?? throw new ApplicationException("End time cannot be null");
```

Comment:
this is actually supported - from the documentation of these properties: 
"
Gets or sets an optional start time in UTC to display in the chart. If not specified, the chart's start time will be based on the alert's time.
"

so we need to duplicate the behavior here

Reply:
Good point, adding a separate task for it

---
In reply to: [304379620](https://github.com/Azure/SmartDetectorsMonitoringAppliance/pull/70#discussion_r304379620) [](ancestors = 304379620)